src/game/text.c

```diff
@@ -406,14 +444,102 @@ static void Text_DrawText(TEXTSTRING *textstring)
+                Output_DrawScreenGradientQuad(
+                    sx, sy, sh / 2, sv / 2, Text_GetMenuColor(MC_PURPLE_E),
+                    Text_GetMenuColor(MC_PURPLE_E),
+                    Text_GetMenuColor(MC_PURPLE_E),
+                    Text_GetMenuColor(MC_PURPLE_C));
+                // Top right
+                Output_DrawScreenGradientQuad(
+                    sx + sh, sy, -sh / 2, sv / 2,
+                    Text_GetMenuColor(MC_PURPLE_E),
+                    Text_GetMenuColor(MC_PURPLE_E),
+                    Text_GetMenuColor(MC_PURPLE_E),
+                    Text_GetMenuColor(MC_PURPLE_C));
+                // Bottom left
+                Output_DrawScreenGradientQuad(
+                    sx, sy + sv, sh / 2, -sv / 2,
+                    Text_GetMenuColor(MC_PURPLE_E),
+                    Text_GetMenuColor(MC_PURPLE_E),
+                    Text_GetMenuColor(MC_PURPLE_E),
+                    Text_GetMenuColor(MC_PURPLE_C));
+                // Bottom right
+                Output_DrawScreenGradientQuad(
+                    sx + sh, sy + sv, -sh / 2, -sv / 2,
+                    Text_GetMenuColor(MC_PURPLE_E),
+                    Text_GetMenuColor(MC_PURPLE_E),
+                    Text_GetMenuColor(MC_PURPLE_E),
+                    Text_GetMenuColor(MC_PURPLE_C));
+            }
+        } else {
+            Output_DrawScreenFBox(sx, sy, sh, sv);
+        }
     }
 
     if (textstring->flags.outline) {
         sx = Screen_GetRenderScale(bxpos);
         sy = Screen_GetRenderScale(bypos);
         sh = Screen_GetRenderScale(bwidth);
         sv = Screen_GetRenderScale(bheight);
-        Output_DrawScreenBox(sx, sy, sh, sv);
+
+        if (g_Config.ui.menu_style == UI_STYLE_PS1) {
+            if (textstring->outline.style == TS_HEADING) {
+                Output_DrawGradientScreenBox(
+                    sx, sy, sh, sv, Text_GetMenuColor(MC_BLACK),
+                    Text_GetMenuColor(MC_BLACK), Text_GetMenuColor(MC_BLACK),
+                    Text_GetMenuColor(MC_BLACK), OUTLINE_THICKNESS);
+            } else if (textstring->outline.style == TS_BACKGROUND) {
+                Output_DrawGradientScreenBox(
+                    sx, sy, sh, sv, Text_GetMenuColor(MC_GREY_TL),
+                    Text_GetMenuColor(MC_GREY_TR),
+                    Text_GetMenuColor(MC_GREY_BL),
+                    Text_GetMenuColor(MC_GREY_BR), OUTLINE_THICKNESS);
+            } else if (textstring->outline.style == TS_REQUESTED) {
+                Output_DrawCentreGradientScreenBox(
+                    sx, sy, sh, sv, Text_GetMenuColor(MC_GREY_E),
+                    Text_GetMenuColor(MC_GREY_C));
+            }
+        } else {
+            Output_DrawScreenBox(
+                sx, sy, sh, sv, Text_GetMenuColor(MC_GOLD_LIGHT),
+                OUTLINE_THICKNESS);
+            Output_DrawScreenBox(
+                sx - 1, sy - 1, sh, sv, Text_GetMenuColor(MC_GOLD_DARK),
+                OUTLINE_THICKNESS);
```

Comment:
Please inline this logic within `DrawScreenBox` and don't draw 2 rectangles here